Client/src/ArmoniK.DevelopmentKit.Client.Common/Submitter/BaseClientSubmitter.cs

```diff
@@ -187,6 +201,8 @@ public IEnumerable<string> SubmitTasksWithDependencies(IEnumerable<Tuple<byte[],
     var serviceConfiguration = SubmitterService.GetServiceConfigurationAsync(new Empty())
                                                .ResponseAsync.Result;
 
+    var payloads = payloadsWithDependencies as ICollection<Tuple<string, byte[], IList<string>>> ?? payloadsWithDependencies.ToList();
```

Comment:
No. If a `ICollection` is required, prefer taking a `ICollection` directly in the API.

Reply:
Here, I don't think ICollection is required. Loosing a little bit of perf in case of retry is not a big deal : in the usual case, it should not happen. However It'd need to be documented.

Reply:
So I removed the ICollection and let the potential multiple enumeation when retry happens